### ISSUE TITLE
How should I continue training?

### ISSUE THREAD

**Issue by ahmadmustafaanis** [ultralytics/ultralytics#2095]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have trained a model for 25 EPOCHS. 
```bash
yolo task=detect mode=train model=yolov8s.pt data=data.yaml  epochs=25 plots=True
```
By seeing the metrics, I do have a feeling that I should train it more. 
![results](https://user-images.githubusercontent.com/47111429/232692752-718f5f5a-ad38-4c23-96bd-c4a4d4298a88.png)

The results are in runs/train3/weights/best.pt

Should I change the model from yolov8s.pt to runs/train3/weights/best.pt or last.pt to continue the training? Or do I have to start the training from scratch?

### Additional

Also can you confirm it if the model needs more training or not?

**Comment by BoscoShin** [comment]
Hi!

u can resume the model. this code.

this code is example.

**model = YOLO("last.pt")
model.train(resume=True)**

**Comment by Laughing-q** [comment]
@ahmadmustafaanis looks you've finished your first training, in this case if you want to train more epochs for better results then you need to launch another new training with:
```bash
yolo task=detect mode=train model=yolov8s.pt data=data.yaml  epochs=50 plots=True
```
50 is just an example, you can set any other number as you want.

**Comment by Laughing-q** [comment]
@ahmadmustafaanis if the training has finished successfully, the code will strip all the remembering stuffs like optimizer states and epochs in the end, so in your case the `last.pt` is a new pretrained weights just like official weights. It'll restart a new training whether you use `last.pt` or official weight as initialized weights, instead of training another 25 epochs to get 50 total epochs.

**Comment by ahmadmustafaanis** [comment]
Thanks

**Comment by glenn-jocher** [comment]
You're welcome, @ahmadmustafaanis! If the training has ended with success, there is no need to train further with the old `last.pt` checkpoint. It's better to start a new training with the default or optimized hyper-parameters or configurations from scratch, and then, after the new training is completed, analyze the validation metrics and freeze the final weights, which will be used in the final predictions or inference on the test data.

**Comment by lovecambi** [comment]
By default lr_scheduler [LambdaLR](https://github.com/ultralytics/ultralytics/blob/243fc4b1fe214ff6c27759dad51c37809db8f7f8/ultralytics/yolo/engine/trainer.py#L245), the lr is related to the total epochs.

**Comment by glenn-jocher** [comment]
@lovecambi hello, Thank you for your interest in YOLOv8! By default, the lr_scheduler LambdaLR is related to the total number of epochs in YOLOv8. However, you can change this default behavior by modifying the scheduling function or replace LambdaLR with a built-in PyTorch scheduler or your custom scheduler. You can find samples and details about the learning rate scheduling techniques for YOLOv8 in the YOLOv8 Github repository in README and the relevant files or classes. Feel free to refer to the official PyTorch documentation or examples for more information or different learning rate scheduling techniques.
